src/network.ts

```diff
@@ -0,0 +1,155 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as networkCheck from 'network-checker';
+import * as os from 'os';
+import * as url from 'url';
+
+import * as constants from './lib/constants';
+import { EEXIST } from './lib/errors';
+import { checkTruthy } from './lib/validation';
+
+import blink = require('./lib/blink');
+
+const networkPattern = {
+	blinks: 4,
+	pause: 1000,
+};
+
+let isConnectivityCheckPaused = false;
+let isConnectivityCheckEnabled = true;
+
+function checkHost(
+	opts: networkCheck.ConnectOptions,
+): boolean | PromiseLike<boolean> {
+	if (!isConnectivityCheckEnabled || isConnectivityCheckPaused) {
+		return true;
+	}
+	return networkCheck.checkHost(opts);
```

Comment:
NIT: this could be
```typescript
return (
    !isConnectivityCheckEnabled ||
    isConnectivityCheckPaused ||
    networkCheck.checkHost(opts)
)
```